src/microbe_stage/MembraneWaterRipple.cs

```
@@ -0,0 +1,452 @@
+            foreach (var propertyName in new[] { "Radius", "radius", "_radius" })
+            {
+                var property = parentMembrane.GetType().GetProperty(propertyName,
+                    BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+
+                if (property != null)
+                {
+                    var value = property.GetValue(parentMembrane);
+                    if (value != null)
+                    {
+                        return Convert.ToSingle(value);
+                    }
+                }
+
+                var field = parentMembrane.GetType().GetField(propertyName,
+                    BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+
+                if (field != null)
+                {
+                    var value = field.GetValue(parentMembrane);
+                    if (value != null)
+                    {
+                        return Convert.ToSingle(value);
+                    }
+                }
+            }
+
+            // Try common getter methods
+            foreach (var methodName in new[] { "GetRadius", "get_Radius", "GetSize", "get_Size" })
+            {
+                var method = parentMembrane.GetType().GetMethod(methodName,
+                    BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+
+                if (method != null && method.GetParameters().Length == 0)
+                {
+                    var value = method.Invoke(parentMembrane, null);
+                    if (value != null)
+                    {
+                        return Convert.ToSingle(value);
+                    }
+                }
+            }
+
+            // Fallback: estimate from scale if membrane is Node3D
+            if (parentMembrane is Node3D node3D)
+            {
+                return node3D.Scale.X * 2.0f;
+            }
+
+            return DEFAULT_MEMBRANE_RADIUS;
+        }
+        catch (Exception)
```

And this silently ignores errors in reading the data??